src/metabase/util/i18n/impl.clj

```diff
@@ -57,14 +57,45 @@
    (when-let [locale (locale locale-or-name)]
      (LocaleUtils/isAvailableLocale locale))))
 
-(defn parent-locale
-  "For langugage + country Locales, returns the language-only Locale. Otherwise returns `nil`.
+(defn- available-locale-names*
```

Comment:
Not sure if it's better to move things around or do `(declare)`

A change from the old implementation is that this now uses a sorted set.

Reply:
Better to move things around. In Clojure it's generally more idiomatic to avoid `declare` if you can and organize stuff so you don't need it